### ISSUE TITLE
changes from matlab unfold

### ISSUE THREAD

**Issue by behinger** [unfoldtoolbox/jlunfold#4]
It is nice for maintenance & coding to keep the structure of unfold and jlunfold similar.

But while implementing some nice-to-have features, I noticed some things are unfortunate:

- [ ] Cross validation of spline numbers is difficult, as the pipeline is modular, but the number of splines are changed in uf_designmat, but the cfg for uf_timeexpanddesignmat is currently not completly saved. This would also require that EEG.events is saved into EEG.unfold.events (not a biggy, but I didn't do it yet; I dislike it, as it doubles datastructures)
- [ ] Different timewindows for different Events is currently not possible in matlab. The outcome would be, that I cannot reshape the EEG.unfold.beta to three dimensions, as the "time" dimension would be different for the different betas. Unsure how to procede here

1) Could be solved by some kind of lazy evaluation, i.e. generate Xdc only once you need it. This is suboptimal in my opinion, as it makes it less transparent & harder to check your EEG.unfold structure
2) not exactly sure. Some type of DataFrame could be useful, which directly encodes what predictor & channel we have there. Have to think about how later matrix products for contrasts etc. intervene here.

**Comment by behinger** [comment]
see #5 for a potential solution. The different ized basis functions might still be a problem though - but I think using tables with submatrices also fixes that (similar to unmixed in matlab)
